homeassistant/components/google/api.py

```diff
@@ -50,12 +49,16 @@ class DeviceAuth(AuthImplementation):
     async def async_resolve_external_data(self, external_data: Any) -> dict:
         """Resolve a Google API Credentials object to Home Assistant token."""
         creds: Credentials = external_data[DEVICE_AUTH_CREDS]
+        delta = creds.token_expiry.replace(tzinfo=datetime.timezone.utc) - dt.utcnow()
```

Comment:
`expires_in` is part of the spec and indicates seconds until the tokens expire. Instead of tracking when we made the request, we should store the time when the tokens actually expire as `expires_at` ([example from OAuth2 helper](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/config_entry_oauth2_flow.py#L92)).  

If we always use `time.time()` (which is epoch time, meaning no timezone), it should work fine:

```suggestion
        delta = creds.token_expiry.timestamp - time.time()
```

Reply:
Aren't you just reverting to what it was before?

Reply:
The point if this change is that I'm saying here that `creds.token_expiry.timestamp() - time.time()` is not correct. That is what it used to do.


To your point about expires_at, the config flow overwrites it.   https://github.com/home-assistant/core/blob/cc807b4d597daaaadc92df4a93c6e30da4f570c6/homeassistant/helpers/config_entry_oauth2_flow.py#L300  -- I did not think it was worth worth to introduce google oauth lib specific complexity into the config flow helpers to fix this. The spec returns expires_in, google's lib gives us expires_at (only), and we're reverting it here to get back expires_in to act like all other oauth clients. This stuff is very brittle, so i think it makes sense to make google calendar only pay the complexity cost.